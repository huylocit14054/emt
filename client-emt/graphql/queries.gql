query getCurrentUser {
  currentUser {
    id
    username
    role
    avatar
  }
}

query getCurrentUserProfile {
  profile: currentUser {
    id
    username
    email
    role
    avatar
  }
}

query getCurrentUserOnClient {
  currentUser @client {
    id
    username
    role
    avatar
  }
}

query getAllUsers {
  users {
    avatar
    id
    username
    email
    createdAt
  }
}

query getProjectOfCurrentUser {
  projectsAsMemberOfCurrentUser {
    id
    name
    description
  }
  projectsAsAdminOfCurrentUser {
    id
    name
    description
  }
}

query getMembersByProjectId($projectId: ID!) {
  projectMembers(projectId: $projectId) {
    id
    role
    user {
      id
      email
      username
      role
      avatar
    }
  }
}

query getProjectById($id: ID!) {
  project(id: $id) {
    id
    name
    description
    isManagedByCurrentUser
  }
}
