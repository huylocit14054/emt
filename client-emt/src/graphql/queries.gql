#import "./fragments/BasicRuleInfo.gql"
#import "./fragments/BasicCompanyInfo.gql"
query getCurrentUser {
  currentUser {
    id
    username
    role
    avatar
  }
}

query getClientCurrentUser {
  currentUser @client {
    id
    username
    role
    avatar
  }
}

query usersSuggestion($query: String!) {
  usersSuggestion(query: $query) {
    id
    username
    email
    avatar
  }
}

query membersSuggestion($query: String!, $projectId: ID!) {
  membersSuggestion(query: $query, projectId: $projectId) {
    id
    username
    email
    avatar
  }
}

query getCurrentUserProfile {
  profile: currentUser {
    id
    username
    email
    role
    avatar
  }
}

query getCurrentUserOnClient {
  currentUser @client {
    id
    username
    role
    avatar
  }
}

query getAllUsers {
  users {
    avatar
    id
    username
    email
    createdAt
  }
}

query getProjectOfCurrentUser {
  projectsAsMemberOfCurrentUser {
    id
    name
    description
    shortenMembers {
      id
      avatar
    }
    memberCount
  }
  projectsAsAdminOfCurrentUser {
    id
    name
    description
    shortenMembers {
      id
      avatar
    }
    memberCount
  }
}

query getMembersByProjectId($projectId: ID!) {
  projectMembers(projectId: $projectId) {
    id
    role
    status
    user {
      id
      email
      username
      role
      avatar
    }
  }
}

query getAssignmentsByProjectId($projectId: ID!) {
  assignments: projectMembers(projectId: $projectId) {
    id
    user {
      id
      username
      avatar
    }
    authorizedDimensions
    authorizations {
      dimension {
        id
      }
      optionAuthorizations {
        option {
          id
          name
          dimension {
            id
          }
        }
      }
    }
  }
  projectDimensions(projectId: $projectId) {
    id
    name
    category
  }
}

query getDimensionsByProjectId($projectId: ID!) {
  projectDimensions(projectId: $projectId) {
    id
    name
    category
    createdAt
    options {
      id
      name
    }
  }
}

query getDimensionSuggestionsByProjectId($projectId: ID!) {
  projectDimensions(projectId: $projectId) {
    id
    name
  }
}

query getProjectDimensionsTreeByProjectId($projectId: ID!) {
  projectDimensionsTree(projectId: $projectId)
}

query getOptionsByDimensionId($dimensionId: ID!) {
  dimensionOptions(dimensionId: $dimensionId) {
    id
    name
  }
}

query getProjectById($id: ID!) {
  project(id: $id) {
    id
    name
    description
    isManagedByCurrentUser
  }
}

query getAssignmentsByMemberId($memberId: ID!) {
  memberAssignments(memberId: $memberId)
}

query getRulesByProjectId($projectId: ID!) {
  projectRules(projectId: $projectId) {
    ...BasicRuleInfo
  }
}

query getAssignmentsOfCurrentMember($projectId: ID!) {
  currentAppliedRule: currentAppliedRuleOfProject(projectId: $projectId) {
    id
    ruleStringToDisplay
  }
  assignments: memberAssignmentsDetails(projectId: $projectId) {
    dimension {
      id
      name
      category
    }
    optionAuthorizations {
      option {
        id
        name
      }
    }
  }
}

query currentMemberUtmHistory($projectId: ID!) {
  currentMemberUtmHistory: memberUtmHistory(projectId: $projectId) {
    id
    url
    createdAt
  }
}

query utmAnalysis($projectId: ID!) {
  utmAnalysis(projectId: $projectId) {
    id
    url
    createdAt
    projectMember {
      id
      user {
        username
        id
        avatar
      }
    }
  }
}

query getAllPlansOfApplication {
  allPlans {
    id
    name
  }
}

query getAllCompaniesOfApplication {
  allCompanies {
    ...BasicCompanyInfo
  }
}

query getCompany($companyId: ID!) {
  company(companyId: $companyId) {
    id
    logo
    name
    currentUserAsMember {
      id
      roles
    }
  }
}
