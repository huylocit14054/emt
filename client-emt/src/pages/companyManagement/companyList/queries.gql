query getCompanyDetails($companyId: ID!, $memberAmount: Int!){
    company(companyId: $companyId){
        id
        status
        name
        description
        email
        address
        phoneNumber
        plan {
            id
            name
        }
        companyMemberCount
        companyMembers(first: $memberAmount) {
            id
            roles
            user {
                id
                username
                avatar
            }
        }
    }
}
