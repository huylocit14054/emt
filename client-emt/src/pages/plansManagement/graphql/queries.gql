#import "./fragments/BasicPlanInfo.gql"

query allPlans{
  allPlans{
    ...BasicPlanInfo
  }
}

query getAllServices {
  services {
    id
    name
  }
}
