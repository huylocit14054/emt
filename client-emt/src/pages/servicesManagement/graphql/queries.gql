#import "./fragments/BasicServiceInfo.gql"
query getAllServices {
  services {
    ...BasicServiceInfo
  }
}