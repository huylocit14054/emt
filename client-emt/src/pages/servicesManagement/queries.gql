query getAllServices {
  services {
    id
    name
    createdAt
    description
  }
}