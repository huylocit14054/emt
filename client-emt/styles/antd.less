@import "~antd/dist/antd.less";

@primary-color: #00b5d0;
mark.highlight-keyword-username {
  font-weight: bold;
  color: #00b5d0;
}

mark.highlight-keyword-email {
  font-weight: bold;
  color: #00b5d0;
}
.small-email-text {
  color: gray;
  margin-left: 5px;
  font-size: 10px;
}

i.add-options-btn {
  &:hover {
    color: @primary-color
  }
}

i.edit-btn {
  &:hover {
    color: @primary-color
  }
}