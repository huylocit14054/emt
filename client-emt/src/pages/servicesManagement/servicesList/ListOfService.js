import React from 'react';
import { Table } from 'antd';
import TimeAgo from 'react-timeago';
import { Query } from 'react-apollo';
import { getAllServices as GET_ALL_SERVICES } from '../graphql/queries.gql';
import ServiceAction from './ServiceAction';

const { Column } = Table;

class ListOfService extends React.Component {
  render() {
    return (
      <Query query={GET_ALL_SERVICES}>
        {({ loading, error, data }) => {
          if (error) return <div>Error! {error.message}</div>;
          return (
            <Table
              dataSource={data.services}
              rowKey="id"
              loading={loading}
              expandedRowRender={service => <p>{service.description}</p>}
              bordered
            >
              <Column title="Name" dataIndex="name" />
              <Column
                title="Created At"
                dataIndex="createdAt"
                render={date => <TimeAgo date={date} />}
              />
              <Column
                title="Action"
                key="action"
                render={(text, record) => (
                  <ServiceAction
                    id={record.id}
                    name={record.name}
                    description={record.description}
                  />
                )}
              />
            </Table>
          );
        }}
      </Query>
    );
  }
}

export default ListOfService;
