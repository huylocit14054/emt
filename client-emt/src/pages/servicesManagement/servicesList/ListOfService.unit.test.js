import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { Table } from 'antd';
import ListOfService from './ListOfService';
import { getAllServices as GET_ALL_SERVICES } from '../graphql/queries.gql';
import wait from '../../../utils/wait';

describe('(Component) Servicelist', () => {
  const mocks = [
    {
      request: {
        query: GET_ALL_SERVICES,
      },
      result: {
        data: {
          services: [
            {
              id: 1,
              name: 'utm',
              description: 'utm service',
              createdAt: Date.now(),
              __typename: 'Service',
            },
            {
              id: 2,
              name: 'oms',
              description: 'oms service',
              createdAt: Date.now(),
              __typename: 'Service',
            },
          ],
        },
      },
    },
  ];
  it('renders...', async () => {
    const wrapper = mount(
      <MockedProvider mocks={[]} addTypename={false}>
        <ListOfService />
      </MockedProvider>
    );
    expect(wrapper).to.have.length(1);
  });

  it('should render loading state initially', () => {
    const wrapper = mount(
      <MockedProvider mocks={[]} addTypename={false}>
        <ListOfService />
      </MockedProvider>
    );
    expect(
      wrapper
        .find('Table')
        .first()
        .props().loading
    ).toBe(true);
  });

  it('should render data', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ListOfService />
      </MockedProvider>
    );
    await wait(0); // wait for responses
    expect(wrapper.text()).toMatchSnapshot();
  });
});
