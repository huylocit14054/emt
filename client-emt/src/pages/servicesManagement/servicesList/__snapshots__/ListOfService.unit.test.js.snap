// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Servicelist should render data 1`] = `"NameCreated AtActionutm0 seconds agoEditDeleteoms0 seconds agoEditDelete1"`;
